### PR TITLE
Modernize cmake KCE-11

### PR DESCRIPTION
**Pull request by paulagb8894**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,17 +20,14 @@
...
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.12)
 
 SET(PROJ_MAIN_NAME "tutorial")
 PROJECT(${PROJ_MAIN_NAME})
 
 
 # Configuration types
 SET(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Configs" FORCE)
-IF(DEFINED CMAKE_BUILD_TYPE AND CMAKE_VERSION VERSION_GREATER "2.8")
```
Comment (mharshe) [inline CMakeLists.txt:31]: Only delete the version check of the if statement. The rest of it should stay.

File: examples/chapter1/CMakeLists.txt [hunk 2]
```diff
@@ -23,66 +23,117 @@ PROJECT(${PROJ_NAME})
...
+  PUBLIC
   ${KLEPSYDRA_CORE_LIBRARIES}
-  ${CMAKE_THREAD_LIBS_INIT}
   atomic
   )
 
+TARGET_INCLUDE_DIRECTORIES(${PROJ_NAME}_1 
+  PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/include 
+  ${KLEPSYDRA_INCLUDE_DIRS}
```
Comment (mharshe) [inline examples/chapter1/CMakeLists.txt:37]: Have you tried deleting this line? It should not be necessary for compilation.

File: examples/chapter1/CMakeLists.txt [hunk 3]
```diff
@@ -23,66 +23,117 @@ PROJECT(${PROJ_NAME})
...
-INCLUDE_DIRECTORIES( ${CMAKE_CURRENT_SOURCE_DIR}/include )
-INCLUDE_DIRECTORIES( ${KLEPSYDRA_INCLUDE_DIRS} )
-
 ADD_EXECUTABLE(${PROJ_NAME}_1 ${${PROJ_NAME}_HEADERS} src/example1.cpp )
 
 TARGET_LINK_LIBRARIES(${PROJ_NAME}_1
+  PUBLIC
   ${KLEPSYDRA_CORE_LIBRARIES}
-  ${CMAKE_THREAD_LIBS_INIT}
   atomic
```
Comment (mharshe) [inline examples/chapter1/CMakeLists.txt:31]: Try deleting the this line too. It should compile.

File: examples/chapter1/CMakeLists.txt [hunk 4]
```diff
@@ -23,66 +23,117 @@ PROJECT(${PROJ_NAME})
...
   )
 
+TARGET_INCLUDE_DIRECTORIES(${PROJ_NAME}_2 
+  PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/include 
+  ${KLEPSYDRA_INCLUDE_DIRS}
```
Comment (mharshe) [inline examples/chapter1/CMakeLists.txt:51]: Same comments here as the previous target. You should not need to put klepsydra_include_dirs anywhere.

**Review by mharshe — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (paulagb8894) [reply CMakeLists.txt:31]: Done

File: examples/chapter1/CMakeLists.txt (hunk 3 above)
Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt:31]: I tried it but it didn't compile

File: examples/chapter1/CMakeLists.txt (hunk 4 above)
Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt:51]: I tried it but it didn't compile too

File: examples/chapter1/CMakeLists.txt (hunk 2 above)
Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt:37]: Yes, I tried everything you told me but It didn't compile withou this line

Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt]: After the kpsr-core update in my PC this works . So, done

File: examples/chapter1/CMakeLists.txt (hunk 3 above)
Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt]: After the kpsr-core update in my PC this works . So, done

File: examples/chapter1/CMakeLists.txt (hunk 4 above)
Comment (paulagb8894) [reply examples/chapter1/CMakeLists.txt]: After the kpsr-core update in my PC this works . So, done

**Review by mharshe — APPROVED** [review]

**Review by pablogh-2000 — APPROVED** [review]

**Pull request merged**